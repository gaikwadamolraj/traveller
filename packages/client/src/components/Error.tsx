import { Alert, AlertIcon } from '@chakra-ui/react'
import type { IErrorProps } from '../dataStructure'

export const Error: React.FC<IErrorProps> = ({ error }) => {
  if (error?.length) {
    return (
      <Alert status="error">
        <AlertIcon />
        {error}
      </Alert>
    )
  }
  return <></>
}
