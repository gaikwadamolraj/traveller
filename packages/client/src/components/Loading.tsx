import { Spinner } from '@chakra-ui/react'
import type { IEmptyProps } from '../dataStructure'

export const Loading: React.FC<IEmptyProps> = () => (
  <Spinner thickness="4px" speed="0.65s" emptyColor="gray.200" color="blue.500" size="xl" />
)
