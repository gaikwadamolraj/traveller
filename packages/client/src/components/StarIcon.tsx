import { StarIcon } from '@chakra-ui/icons'
import type { StarIconProps } from '../dataStructure'

export const StarIconWithAction: React.FC<StarIconProps> = ({ page, city, color, handleFunc }) => {
  if (typeof handleFunc === 'function') {
    return <StarIcon onClick={() => handleFunc(city, page)} marginLeft={'20px'} color={color} />
  }

  return <StarIcon marginLeft={'20px'} color={color} />
}
