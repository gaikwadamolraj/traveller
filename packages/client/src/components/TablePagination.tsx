import * as React from 'react'
import { Table } from 'react-chakra-pagination'
import { Box, useColorMode } from '@chakra-ui/react'
import { FiDatabase } from 'react-icons/fi'
import type { ICity, TableProps } from '../dataStructure'
import { CONSTANTS } from '../dataStructure'
import { StarIconWithAction } from './StarIcon'

const tableColumns = [
  {
    Header: 'Name',
    accessor: 'name' as const,
  },
  {
    Header: 'Country',
    accessor: 'country' as const,
  },
  {
    Header: 'Wish List',
    accessor: 'wishAction' as const,
  },
  {
    Header: 'Visited',
    accessor: 'visitAction' as const,
  },
]

export const TablePagination: React.FC<TableProps> = ({
  pageNumber = CONSTANTS.DEFAULT_PAGE,
  cities,
  handleWishlist,
  handleVisited,
}: TableProps) => {
  const [page, setPage] = React.useState(pageNumber)
  const { colorMode } = useColorMode()

  const tableData = cities?.map((city: ICity) => ({
    name: city.name,
    country: city.country,
    wishAction: (
      <StarIconWithAction
        page={page}
        city={city}
        color={city.wishlist ? 'green' : 'gray'}
        handleFunc={handleWishlist}
      />
    ),
    visitAction: (
      <StarIconWithAction page={page} city={city} color={city.visited ? 'green' : 'gray'} handleFunc={handleVisited} />
    ),
  }))

  return (
    <Box>
      <Box>
        <Table
          key={'paginationTable'}
          colorScheme={colorMode === 'light' ? 'blue' : 'yellow'}
          emptyData={{
            icon: FiDatabase,
            text: 'No city found',
          }}
          totalRegisters={cities?.length}
          page={page}
          onPageChange={(page: number) => setPage(page)}
          columns={tableColumns}
          data={tableData}
        />
      </Box>
    </Box>
  )
}
