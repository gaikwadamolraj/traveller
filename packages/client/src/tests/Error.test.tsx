import React from 'react'
import { screen } from '@testing-library/react'
import { render } from '../test-utils'
import {Error} from '../components/Error'

describe('<Error /> component', () => {
  it('With message', () => {
    render(<Error error="amol" />)
    const HeadingComponent = screen.getByText(/^amol$/i)
    expect(HeadingComponent).toBeInTheDocument()
  })
  it('Blank render for error', () => {
    const { container } = render(<Error error="" />)
    expect(container).toMatchSnapshot()
  })
})
