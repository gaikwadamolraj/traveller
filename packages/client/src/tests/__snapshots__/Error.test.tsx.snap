// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> component Blank render for error 1`] = `<div />`;
